### PR TITLE
[Testing] Added Camel unit tests

### PR DESCRIPTION
**Pull request by markoer**
Added tests:
PayloadFactory
DefaultCamelCloudService sequence
KuraCloudComponentResolver
KuraCloudComponent
KuraServiceFactory
KuraCloudProducer

Also merged project files and moved existing integration tests

### CODE REVIEW

**Review by MMaiero — APPROVED** [review]
Fine for me. But let's wait @ctron before merging.

File: kura/test/org.eclipse.kura.camel.test/build.properties [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
+#   are made available under the terms of the Eclipse Public License v1.0
+#   which accompanies this distribution, and is available at
+#   http://www.eclipse.org/legal/epl-v10.html
+################################################################################
+
+source.. = src/main/java
 output.. = target/classes
 bin.includes = META-INF/,\
-               .,\
-               about.html
```
Comment (ctron) [inline kura/test/org.eclipse.kura.camel.test/build.properties]: Each bundle must have an "about.html"

**Review by ctron — COMMENTED** [review]
THB, I have no clue what most of the tests actually try to do. I can't really make a decision here.

**Comment by markoer** [comment]
The tests provide detailed coverage of the code. Some of the tested scenarios are the ones you didn't already cover in integration tests.

**Pull request merged**